src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -120,6 +120,9 @@ public async Task Dispatch(TransportOperations outgoingMessages, TransportTransa
                 var operations = destinationAndOperations.Value;
 
                 var messagesToSend = new Queue<ServiceBusMessage>(operations.Count);
+                // We assume the majority of the messages will be batched and only a few will be sent individually
+                // and therefore it is OK in those rare cases for the list to grow.
+                var messagesThatCouldntBeSent = new List<ServiceBusMessage>(0);
```

Comment:
messagesThatCouldntBeSent  => messagesThatCouldntBeBatched ?

Reply:
I would go with `messagesThatCouldNotBeBatched`

Reply:
Will adjust

Reply:
I think messagesTooLargeToBeBatched would have even be better looking at what the intent is